prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -341,41 +341,61 @@ class DetailedSubmissionHistory(
                 // Check if destination has already been added
                 // if it is increase item counts
                 // otherwise add it to destinations
-                val index = destinations.indexOf(descendantDest)
-                if (index >= 0) {
-                    destinations[index].itemCount += descendantDest.itemCount
-                    destinations[index].itemCountBeforeQualFilter =
-                        destinations[index].itemCountBeforeQualFilter?.plus(
-                            descendantDest.itemCountBeforeQualFilter ?: 0
-                        )
-                } else {
-                    destinations += descendantDest
+                with(destinations) {
+                    val index = indexOfFirst {
+                        it.organizationId == descendantDest.organizationId && it.service == descendantDest.service
+                    }
+                    if (index >= 0) {
+                        get(index).itemCount += descendantDest.itemCount
```

Comment:
Don't repeat get(index). Set it to a variable or replace the index with a filter

Reply:
Using firstOrNull now.